recipes/dotnet_agent.rb

```diff
@@ -0,0 +1,86 @@
+agent = node['appdynamics']['dotnet_agent']
+controller = node['appdynamics']['controller']
+proxy = node['appdynamics']['http_proxy'] 
+bitness = node['kernel']['os_info']['os_architecture']
+install_directory = agent['install_dir']
+system_direcotry = node['kernel']['os_info']['windows_directory']
+
+agent_msi = ""
+##################################################################################################################################
+# The below paths can also be used instead of temp direcotry. But there will be msiexec exceptions if there is space in the path #
+# agent_msi_path = "#{Chef::Config[:file_cache_path]"																			 #
+# setup_config = "#{Chef::Config[:file_cache_path]}\\setup.xml"																	 #
+# install_log_file = "#{Chef::Config[:file_cache_path]}\\DotnetAgentInstall.log"												 #
+##################################################################################################################################
+agent_msi_path = "#{system_direcotry}\\Temp"
+setup_config = "#{agent_msi_path}\\setup.xml"
+install_log_file = "#{agent_msi_path}\\DotnetAgentInstall.log"
+
+
+#Check the bitness of the OS to determine the installer to download and run
+case bitness
+when /64/
+ agent_msi = "#{agent_msi_path}\\dotNetAgentSetup64.msi"
+ source_path = "#{agent['source']}/dotNetAgentSetup64.msi"
+when /32/
+ agent_msi = "#{agent_msi_path}\\dotNetAgentSetup.msi"
+  source_path = "#{agent['source']}/dotNetAgentSetup.msi"
+else
+ agent_msi = "Unsupported OS bitness"
+end
+
+#Download the msi file from source
+remote_file "#{agent_msi}" do
+  source source_path
+  checksum agent['checksum']
+end
+
+#Environment Validation
+# Comment if you're installing 4.1
```

Comment:
We still have to confirm this with the Devs. I just added that as a comment. Matt has to come back on that.